### PR TITLE
Fix leftover operation files in CC tests

### PR DESCRIPTION
**Pull request by blindpirate**
[Original discussion](https://gradle.slack.com/archives/CH43P3A10/p1636104331055900)

We suspect that some build operation tracing system properties were set in test A but not cleaned up properly, resulting in test B generating leftover trace files. This PR captures all system properties before the invocation, and restore them after.

### CODE REVIEW

File: subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/fixtures/ToolingApiBackedGradleExecuter.groovy [hunk 1]
```diff
@@ -53,13 +53,19 @@ class ToolingApiBackedGradleExecuter extends AbstractGradleExecuter {
...
         args.remove("--no-daemon")
 
-        usingToolingConnection(workingDir) { connection ->
-            connection.newBuild()
-                .withArguments(args)
-                .addJvmArguments(jvmArgs)
-                .setStandardOutput(new TeeOutputStream(output, System.out))
-                .setStandardError(new TeeOutputStream(error, System.err))
-                .run()
+        def systemPropertiesBeforeInvocation = new HashMap<String, Object>(System.getProperties())
```
Comment (wolfs) [inline subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/fixtures/ToolingApiBackedGradleExecuter.groovy:56]: I wonder if we need to move this into `usingToolingConnection`, given that the current system property cleanup is in there. And we probably only should mess with the system properties when running using the embedded executer.

What we should do for sure is to remove the current system property cleanup: https://github.com/gradle/gradle/blob/20f814d3f0e26eb080a73e0cac19e7ddf602f226/subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/fixtures/ToolingApiBackedGradleExecuter.groovy#L92-L95

**Review by wolfs — APPROVED** [review]
LGTM!

File: subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/fixtures/ToolingApiBackedGradleExecuter.groovy [hunk 2]
```diff
@@ -84,14 +84,17 @@ class ToolingApiBackedGradleExecuter extends AbstractGradleExecuter {
...
+
+        def systemPropertiesBeforeInvocation = new HashMap<String, Object>(System.getProperties())
+
         try {
             toolingApi.withConnection {
                 action.execute(it)
             }
         } finally {
             if (GradleContextualExecuter.embedded) {
-                System.clearProperty(StartParameterBuildOptions.ConfigurationCacheOption.PROPERTY_NAME)
```
Comment (wolfs) [inline subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/fixtures/ToolingApiBackedGradleExecuter.groovy:93]: There is now a checkstyle warning about an unused import in this class.

**Comment by blindpirate** [comment]
Thanks @wolfs 

@bot-gradle test and merge

**Pull request merged**